storage/shard.go

```diff
@@ -1613,15 +1609,22 @@ func (s *dbShard) Bootstrap(
 		if entry == nil {
 			// Synchronously insert to avoid waiting for
 			// the insert queue potential delayed insert
-			entry, err = s.insertSeriesSync(dbBlocks.ID,
-				ident.NewTagSliceIterator(dbBlocks.Tags),
+			tagsIter := s.identifierPool.TagsIterator()
+			tagsIter.Reset(dbBlocks.Tags)
+			entry, err = s.insertSeriesSync(dbBlocks.ID, tagsIter,
 				insertSyncIncReaderWriterCount)
 			if err != nil {
 				multiErr = multiErr.Add(err)
 				continue
 			}
 		}
 
+		// No longer require tags as we copy them in insert series sync
+		// or if we found the series then we don't require them for insertion
+		// at all
+		dbBlocks.Tags.Finalize()
```

Comment:
why not the id too?

Reply:
Unfortunately in some cases the ID is kept ref to (with series cache policy `all_metadata`).

Reply:
Could you put down a note explaining that, and that we can delete it once we trash that policy.